tests/phpunit/integration/Core/Authentication/Google_ProxyTest.php

```diff
@@ -218,20 +334,24 @@ public function test_exchange_site_code() {
 	 * given URL will yield a specific response.
 	 *
 	 * @param string $request_url   Request URL to modify response for.
-	 * @param array  $response_data Response data to return for the request. Will be JSON-encoded.
+	 * @param (array)  $response_data Response data to return for the request. Will be JSON-encoded if it is an array.
 	 * @param int    $response_code Optional. Response status code to return. Default 200.
 	 */
-	private function mock_http_request( $request_url, array $response_data, $response_code = 200 ) {
+	private function mock_http_request( $request_url, $response_data, $response_code = 200 ) {
 		add_filter(
 			'pre_http_request',
 			function ( $preempt, $args, $url ) use ( $request_url, $response_data, $response_code ) {
 				if ( $request_url !== $url ) {
 					return $preempt;
 				}
 
+				if ( is_array( $response_data ) ) {
+					$response_data = json_encode( $response_data );
+				}
+
 				return array(
 					'headers'       => array(),
-					'body'          => json_encode( $response_data ),
+					'body'          => $response_data,
```

Comment:
Why was this changed? I don't see any usage where `$response_data` is not an array, and I think it's better to keep the signature consistent, especially if there's no reason to change it.

Reply:
In a previous commit, I updated this function to allow tests to pass a WP_Error via $response_data before splitting it out later into `mock_http_failure`.

I've added this type back in. 👌